apps/extension/src/core/domains/accounts/handler.ts

```diff
@@ -45,63 +37,50 @@ import { SOURCES } from "../mnemonics/store"
 import { AccountsCatalogData, emptyCatalog } from "./store.catalog"
 
 export default class AccountsHandler extends ExtensionHandler {
-  private async accountCreate({ name, type, mnemonicId }: RequestAccountCreate): Promise<string> {
+  private async accountCreate({ name, type, ...options }: RequestAccountCreate): Promise<string> {
     const password = this.stores.password.getPassword()
     assert(password, "Not logged in")
 
     const allAccounts = keyring.getAccounts()
     const existing = allAccounts.find((account) => account.meta?.name === name)
     assert(!existing, "An account with this name already exists")
 
-    const rootAccount = mnemonicId ? getRootAccountForMnemonic(mnemonicId, type) : false
-
-    let seed = mnemonicGenerate()
-    let derivedMnemonicId = mnemonicId
-    if (mnemonicId) {
-      const seedResult = await this.stores.seedPhrase.getSeed(mnemonicId, password)
-      if (seedResult.err) throw seedResult.val
-      if (!seedResult.val) throw new Error("Seed not stored locally")
-      seed = seedResult.val
+    let derivedMnemonicId: string
+    if ("mnemonicId" in options) {
+      derivedMnemonicId = options.mnemonicId
     } else {
-      const newSeedId = await this.stores.seedPhrase.add(
-        "Talisman Recovery Phrase",
-        seed,
+      const newMnemonicId = await this.stores.seedPhrase.add(
+        `${name} Recovery Phrase`,
+        options.mnemonic,
         password,
-        SOURCES.Generated
+        SOURCES.Generated,
+        options.confirmed
       )
-      if (newSeedId.err) throw new Error("Unable to store new seed")
-      derivedMnemonicId = newSeedId.val
+      if (newMnemonicId.err) throw new Error("Failed to store new mnemonic")
+      derivedMnemonicId = newMnemonicId.val
     }
 
-    const accountMeta = { derivedMnemonicId }
-    if (!rootAccount) {
-      const { pair } = keyring.addUri(seed, password, {
-        name,
-        origin: AccountTypes.TALISMAN,
-        ...accountMeta,
-      })
-      talismanAnalytics.capture("account create", { type, method: "parent" })
-      return pair.address
-    } else {
-      const { val: derivationPath, err } = getNextDerivationPathForMnemonic(seed, type)
-      if (err) throw new Error(derivationPath)
+    const mnemonicResult = await this.stores.seedPhrase.getSeed(derivedMnemonicId, password)
```

Comment:
It seems like this should go in the `if ("mnemonicId" in options)` block above, since in the other case, we know whether or not we are able to successfully store the mnemonic